co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureInstance.java

```diff
@@ -131,22 +141,79 @@ public void setRxMessagePort(int rxMessagePort) {
     }
 
     /**
-     * Returns the time synchronization port of the infrastructure node
+     * Returns the time synchronization port of the infrastructure instance
      * 
-     * @return int the time synchronization port of the infrastructure node
+     * @return int the time synchronization port of the infrastructure instance
      */
     public int getTimeSyncPort() {
         return timeSyncPort;
     }
 
     /**
-     * Sets the time synchronization port of the infrastructure node
+     * Sets the time synchronization port of the infrastructure instance
      * 
      * @param timeSyncPort the port to set
      */
     public void setTimeSyncPort(int timeSyncPort) {
         this.timeSyncPort = timeSyncPort;
     }
+    
+    /**
+     * Returns the simulated interaction port of the infrastructure instance.
+     * 
+     * @return int simulated interaction port of the infrastructure instance.
+     */
+    public int getSimulatedInteractionPort() {
+        return simulatedInteractionPort;
+    }
+
+
+    /**
+     * Sets the simulated interaction port of the infrastructure instance.
+     * 
+     * @param simulatedInteractionPort int simulated interaction port of the infrastructure
+     * instance.
+     */
+    public void setSimulatedInteractionPort(int simulatedInteractionPort) {
```

Comment:
Maybe check if input to make sure it doesn't receive negative values. The same comment for other setter functions to update port numbers if use int data type.

Reply:
The more we change these methods, the more code coverage they require. The setter is actually never used anywhere in the code so I do not think it is worth the extra effort here. We also do not do this validation on the infrastructure registration so doing it here may not make much sense